lib/systems/parse.nix

```diff
@@ -421,6 +421,61 @@ rec {
 
   ################################################################################
 
+  types.openCpuModel = mkOptionType {
+    name = "cpu-model";
+    description = "cpu model";
+    merge = mergeOneOption;
+  };
+
+  types.cpuModel = cpu: enum (attrValues (cpuModels cpu));
+
+  cpuModels = cpu: let
+    mkCpuModel = { zig ? "baseline", gnu ? "generic", llvm ? "generic" }@args:
+      { zig = "baseline"; gnu = "generic"; llvm = "generic"; } // args;
+    mkSimpleCpuModel = name:
+      mkCpuModel {
+        llvm = name;
+        gnu = name;
+        zig = name;
+      };
+    generic = mkCpuModel {};
+  in setTypes types.openCpuModel ({
+    aarch64 = {
+      ampere1 = mkCpuModel {
+        llvm = "ampere1";
+        zig = "ampere1";
+      };
+      ampere1a = mkCpuModel {
+        llvm = "ampere1a";
+        zig = "ampere1a";
+      };
+      apple_m1 = mkCpuModel {
+        llvm = "apple-m1";
+        zig = "apple_m1";
+      };
+      apple_m2 = mkCpuModel {
+        llvm = "apple-m2";
+        zig = "apple_m2";
+      };
+      cortex_a53 = mkCpuModel {
+        llvm = "cortex-a53";
+        gnu = "cortex-a53";
+        zig = "cortex_a53";
+      };
```

Comment:
So far all of these follow the same pattern, of being the same apart from underscores vs hyphens.  Are there counterexamples?

Reply:
That's true though in the case of Apple Silicon, the model isn't supported by GCC yet. Should I make a simple wrapper which generates the names from a single name?

Reply:
Might be able to get rid of the `zig` one and just replace `-` with `_` at least.

Reply:
I made some adjustments so now the function to make the CPU model definition simply can be used for most systems. I decided to keep `zig` in because of `baseline`.